Ignore disabled creators

fnibbit opened the pull request:
Modifies the "search" endpoint of the creator authority to return only active creators, which results in their exclusion from autocomplete suggestions.  The "find/show" endpoint is unmodified.  Connected to https://github.com/MPLSFedResearch/cypripedium/issues/427

fnibbit committed:
Exclude inactive creators from autocomplete suggestions

* Filter out inactive creators in creator authority search

* Set locale in creator model spec to address order-dependent failures

fnibbit committed:
Style check

maxkadel commented on spec/models/creator_spec.rb:
```diff
@@ -2,6 +2,9 @@
 require 'rails_helper'
 
 RSpec.describe Creator, type: :model do
+  before do
```
I'm curious why this is needed?

fnibbit replied:
Order-dependent failures on this line: `end.to raise_error(ActiveRecord::RecordInvalid, /Validation failed: Viaf id already in the system/)`, where sometimes it would give the RecordInvalid error with a "translation missing" instead of that message.  There's probably something better that could/would/should be done about this one? but that's what I got to work.

mark-dce replied:
It might be good to pull this out into it's own PR then and past the failing test.  Then if anyone asks about this in the future, we can use git blame and we'll see the PR about the flapping test.

maxkadel replied:
Broke this out into its own issue - @fnibbit if you have a seed that this happens on that would be a helpful thing to add to the ticket. https://github.com/MPLSFedResearch/cypripedium/issues/482

maxkadel committed:
Extend authority request spec to include active-ness

maxkadel reviewed (approved):
Looks good! I'm curious just personally about the i18n, but can be merged without answering my curiosity :)

mark-dce commented on spec/requests/creator_authority_request_spec.rb:
```diff
@@ -5,23 +5,25 @@
+      { "id": "https://ideas.repec.org/e/pca42.html", "label": "Carlstrom, Charles T., 1960-", "active": true },
+      { "id": "https://ideas.repec.org/f/pca205.html", "label": "Caselli, Francesco, 1966-", "active": true },
+      { "id": "https://ideas.repec.org/e/pca73.html", "label": "Caucutt, Elizabeth M. (Elizabeth Miriam)", "active": true },
+      { "id": "https://ideas.repec.org/f/pca963.html", "label": "Cavalcanti, Ricardo de Oliveira", "active": true }
```
It feels like this is a lot of setup and maybe obscures the specific creator used in the test.

Do we need more than 4 creators for this test?  i.e. is there a way it might break with 12 that it wouldn't with fewer?

Would it also make sense to use the creator factory instead of manually constructing them.
```
let!( other_creators) { 3.times do { FactoryBot.create(:creator. active: true) }
let!( my_creator ) { FactoryBot.create(:creator. display_name:  "Cagetti, Marco", active: true) }
```

mark-dce replied:
We'll leave using the factory for a future ticket.

mark-dce commented on spec/requests/creator_authority_request_spec.rb:
```diff
@@ -74,7 +76,7 @@
       json_body = JSON.parse(response.body)
-      expect(json_body.count).to eq 13
+      expect(json_body.count).to eq 14
       expect(json_body.first["label"]).to eq "Cagetti, Marco"
```
Unless the order is important here, we might want to change this to test for inclusion instead of matching on the first element:
```
expect(json_body).to include_json( label: "Cagetti, Marco")
```

see https://relishapp.com/waterlink/rspec-json-expectations/docs/json-expectations/include-json-matcher#expecting-json-string-to-include-simple-json

maxkadel replied:
Yes, order is important here (we expect them to be returned in alphabetical order).

mark-dce reviewed (changes requested):
I think we should pull out the code for the flaky test into another PR.

We might also want to use the Creator factory instead of doing all setup manually.

maxkadel committed:
Clean up tests

- Take out setup to address order-dependent failure - has its own ticket 
now
- Simplify setup for request spec

Co-Authored-By: Mark Bussey <mark@curationexperts.com>
